File: data/imported/knowledge/slides/net/merged/snippets/snippet_089.cs
[Fact]
    public void TestCloneSlide()
    {
        // add_clone duplicates a slide with its shapes.
        using var pres = new Presentation();
        var slide = pres.Slides[0];
        slide.Shapes!.AddAutoShape(ShapeType.Rectangle, 50, 50, 200, 100);
        pres.Slides.AddClone(slide);
        pres.Slides.Count.Should().Be(2);
        pres.Slides[1].Shapes!.Count.Should().BeGreaterThanOrEqualTo(1);
    }

File: data/imported/knowledge/slides/cpp/merged/snippets/snippet_089.cpp
TEST_F(TextIntegrationTest, AddPortion) {
    Presentation pres;
    auto& shape = pres.slides()[0].shapes().add_auto_shape(
        ShapeType::RECTANGLE, 50, 50, 400, 100);
    shape.text_frame()->set_text("Hello ");
    Portion new_portion("World!");
    shape.text_frame()->paragraphs()[0].portions().add(std::move(new_portion));
    EXPECT_NE(shape.text_frame()->text().find("World!"), std::string::npos);
}